torchsynth/synth.py

```diff
@@ -214,15 +214,25 @@ def randomize(self, seed: Optional[int] = None):
         Randomize all parameters
         """
         if seed is not None:
-            # Profile to make sure this isn't too slow?
-            mt19937_gen = csprng.create_mt19937_generator(seed)
+            self.generator.manual_seed(seed)
             for parameter in self.parameters():
                 if not ModuleParameter.is_parameter_frozen(parameter):
-                    parameter.data.uniform_(0, 1, generator=mt19937_gen)
+                    # TODO figure out how to support different batch sizes and
+                    # maintain reproducibility.
```

Comment:
Okay yeah a little thorny. So we can close this and move on and fix it later, make an issue, put it in the comments, and call it a day

Reply:
I have some ideas how to do it but let's just make an issue and put it in the code and do it later